tools/sdk-generation-pipeline/packages/sdk-generation-cli/src/cli/dockerCli/core/DockerContext.ts

```diff
@@ -0,0 +1,102 @@
+import { Logger } from "winston";
+import { DockerCliConfig } from "../schema/dockerCliConfig";
+import { initializeLogger } from "@azure-tools/sdk-generation-lib";
+import path from "path";
+import fs from "fs";
+import { sdkToRepoMap } from "./generateCodesInLocal";
+
+export class DockerContext {
+    mode: 'generateCodesInLocal' | 'growUp' | 'generateCodesInPipeline';
+    readmeMdPath?: string;
+    tag?: string;
+    sdk: string[];
+    specRepo?: string;
+    workDir?: string;
+    sdkRepo?: string;
+    resultOutputFolder?: string;
+    logger: Logger;
+
+    /*
+    * there are different modes to use the docker image:
+    * 1. local: generate codes
+    * 2. local: grow up
+    * 3. pipeline: generate codes
+    * */
+    public initialize(inputParams: DockerCliConfig) {
```

Comment:
Is this context used for generating SDK for one language or for all languages?
There is one sdkRepo but sdk seems to be a list.

Reply:
the sdk list is used by generating codes in local. The sdk-repo is used by generating in pipeline and there is only one sdk in this scenario. We need to handle different scenario here.